### PR TITLE
[AutoDiff] Replace SIMD use of @differentiable(jvp:vjp:) with @derivative(of:)

### PR DESCRIPTION
**Pull request by bartchr808**
With the change of using `@derivative(of:)` as done in [this PR](https://github.com/tensorflow/swift-apis/pull/591#event-2903053279), I started off with some parts of [TF-1085](https://bugs.swift.org/browse/TF-1085) by updating SIMD.

### CODE REVIEW

File: stdlib/public/core/SIMDVector.swift [hunk 1]
```diff
@@ -90,13 +90,6 @@ extension SIMD {
   
   /// A vector with the specified value in all lanes.
   @_transparent
-  // SWIFT_ENABLE_TENSORFLOW
-  @differentiable(vjp: _vjpInit(repeating:)
```
Comment (dan-zheng) [inline stdlib/public/core/SIMDVector.swift]: Could you please keep `@differentiable` attributes but remove the `vjp: xxx` argument? `@differentiable` attributes on public original declarations ensures that derivatives are publicly accessible.

This has been done consistently in other `@differentiable(jvp:vjp:) -> @derivative` PRs.

Comment (bartchr808) [reply stdlib/public/core/SIMDVector.swift]: Oh okay sure!

**Comment by bartchr808** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
`ParseableInterface/verify_stdlib.swift` fails due to TF-1094:
```
/Users/swiftci/jenkins/workspace/swift-PR-TensorFlow-macOS/Ninja-ReleaseAssert+stdlib-Release/swift-macosx-x86_64/lib/swift/macosx/Swift.swiftmodule/x86_64.swiftinterface:23857:4: error: function is not differentiable
  @differentiable(wrt: value where Self : Swift.Differentiable, Self == Self.TangentVector, Self.Scalar : Swift.BinaryFloatingPoint, Self.Scalar : Swift.Differentiable, Self.Scalar == Self.Scalar.TangentVector)
  ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/swiftci/jenkins/workspace/swift-PR-TensorFlow-macOS/Ninja-ReleaseAssert+stdlib-Release/swift-macosx-x86_64/lib/swift/macosx/Swift.swiftmodule/x86_64.swiftinterface:23858:24: note: when differentiating this function definition
  @_transparent public init(repeating value: Self.Scalar) {
                       ^
/Users/swiftci/jenkins/workspace/swift-PR-TensorFlow-macOS/Ninja-ReleaseAssert+stdlib-Release/swift-macosx-x86_64/lib/swift/macosx/Swift.swiftmodule/x86_64.swiftinterface:23860:14: note: cannot differentiate through a non-differentiable result; do you want to use 'withoutDerivative(at:)'?
    for i in indices { self[i] = value }
             ^
<unknown>:0: error: failed to load module 'Swift'
```

The reason is `SIMD.init(repeating:)` is serialized but `SIMD._vjpInit(repeating:)` is not. This causes `SIMD._vjpInit(repeating:)` to not be registered as a derivative during `.swiftinterface` compilation.

A workaround is to mark `SIMD._vjpInit` as `@inlinable` to make it serialized. We can visit a robust solution for TF-1094 later.

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow macOS

**Review by rxwei — APPROVED** [review]

**Review by dan-zheng — APPROVED** [review]

**Pull request merged**